src/sql/src/func.rs

```diff
@@ -3147,6 +3148,85 @@ pub static MZ_INTERNAL_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(
             // message is the second argument.
             params!(Any, String) => VariadicFunc::ErrorIfNull => Any, oid::FUNC_MZ_ERROR_IF_NULL_OID;
         },
+        "mz_name_to_global_id" => Table {
+            // If the user wants to specify a database, we also require them to specify which
+            // schemas in that database they want us to search in that database.
+            params!(String, ParamType::Plain(ScalarType::Array(Box::new(ScalarType::String))), String) => sql_impl_table_func("
+                SELECT o.id
+                FROM
+                    mz_catalog.mz_objects AS o
+                        JOIN mz_catalog.mz_schemas AS s ON o.schema_id = s.id
+                WHERE
+                    o.name = $3::pg_catalog.text AND s.name =ANY ($2::pg_catalog.text[])
+                        AND
+                    (
+                        s.database_id IS NULL
+                            OR
+                        s.database_id
+                        = (SELECT id FROM mz_catalog.mz_databases WHERE name = $1::pg_catalog.text)
+                    )
+            ") => ReturnType::set_of(String.into()), oid::FUNC_MZ_NAME_TO_GLOBAL_ID_FULL_QUAL;
```

Comment:
It might be unnecessary, but it would probably be useful to order the results by schema index in `$2`. That way the results come back in the order that they would be resolved through normal name resolution. I don't think that's required for this PR though, just thought I'd call it out.

Reply:
Sadly unnest cannot preserver the order of the input (at least not without a substantial refactor), so we don't have a means of providing this ordering AFAICT.